gateway/src/apicast/upstream.lua

```diff
@@ -196,6 +196,17 @@ end
 function _M:set_skip_https_connect_on_proxy()
   self.skip_https_connect = true
 end
+
+
+function _M:set_keepalive_key_if_https(context)
```

Comment:
I think it's a big strange to find an "if_something" in a function name. I find it more natural to have the if outside the function.